From 225bd1de16fb2e330be5bad261d427d00293c55e Mon Sep 17 00:00:00 2001
From: Sylwester Nawrocki <snawrocki@kernel.org>
Date: Wed, 16 Aug 2017 22:31:01 +0200
Subject: [PATCH] ARM: s3c24xx: Fix NAND ECC mode for mini2440 board
Git-commit: 225bd1de16fb2e330be5bad261d427d00293c55e
Patch-mainline: v4.14-rc1
References: bsc#1051510

After commit e9f66ae23c20 ("mtd: s3c2410: make ecc mode configurable via
platform data") booting of the mini2440 board is broken, since
for this board we have CONFIG_MTD_NAND_S3C2410_HWECC set. The console
gets flooded with errors reported by the mtd subsystem.
This patch changes ecc_mode to NAND_ECC_HW which restores proper
operation of the NAND flash memory.

Cc: Sergio Prado <sergio.prado@e-labworks.com>
Fixes: e9f66ae23c20 ("mtd: s3c2410: make ecc mode configurable via platform data")
Signed-off-by: Sylwester Nawrocki <snawrocki@kernel.org>
Signed-off-by: Krzysztof Kozlowski <krzk@kernel.org>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 arch/arm/mach-s3c24xx/mach-mini2440.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-s3c24xx/mach-mini2440.c b/arch/arm/mach-s3c24xx/mach-mini2440.c
index 71af8d2fd320..ff0adcdf1dc1 100644
--- a/arch/arm/mach-s3c24xx/mach-mini2440.c
+++ b/arch/arm/mach-s3c24xx/mach-mini2440.c
@@ -287,7 +287,7 @@ static struct s3c2410_platform_nand mini2440_nand_info __initdata = {
 	.nr_sets	= ARRAY_SIZE(mini2440_nand_sets),
 	.sets		= mini2440_nand_sets,
 	.ignore_unset_ecc = 1,
-	.ecc_mode       = NAND_ECC_SOFT,
+	.ecc_mode       = NAND_ECC_HW,
 };
 
 /* DM9000AEP 10/100 ethernet controller */
-- 
2.18.0

